### PR TITLE
Update API to support template-coq

### PR DESCRIPTION
**Pull request by gares**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
I don't have any objections.
(because none of the proposals violate any of the API [properties](https://github.com/coq/coq/blob/trunk/API/PROPERTIES) we currently maintain).

**Comment by JasonGross** [comment]
Should we also add `template-coq` to the ci?

File: API/API.ml [hunk 1]
```diff
@@ -180,7 +180,27 @@ module Prelude =
    there is no "kernel/entries.ml" file *)
 module Entries =
   struct
-    type mutual_inductive_entry = Entries.mutual_inductive_entry
+  type local_entry = Entries.local_entry =
```
Comment (JasonGross) [inline API/API.ml]: Is the indentation here off from the rest of the file?

Comment (matejkosik) [inline API/API.ml]: I'll try to [fix formatting](https://github.com/matejkosik/coq/blob/6c384c49404ee5ed3af1679a76ad2aeb4d383dc2/API/PROBLEMS/uncategorized#L51) when there will be a good opportunity (no PRs that would be broken).

**Comment by herbelin** [comment]
For the record, in the same line, porting the plugins in user contribs, aac-tactics, containers, etc. would require adding things to API.mli, or alternatively, to changing the code.

**Comment by gares** [comment]
Close, there is not API file now

**Pull request closed without merging**